model/host/host.go

```diff
@@ -810,12 +810,16 @@ func (h *Host) SetProvisioning(ctx context.Context) error {
 }
 
 // SetDecommissioned sets the host as decommissioned. If checkTaskGroup is set,
-// we only update the host if it hasn't started running a task group.
-func (h *Host) SetDecommissioned(ctx context.Context, user string, checkTaskGroup bool, logs string) error {
+// we only update the host if it hasn't started running a task group. If terminateIfBusy
+// isn't set, only update the host if there is no running task.
```

Comment:
Do we need two separate flags, one to check the running task group and one to check the running task itself? I thought `checkTaskGroup` was a way to more gracefully decommission hosts that aren't busy doing something, and checking for any running task is a more general check than checking for a task group. Just reading through all the usages of `SetDecommissioned`, the `checkTaskGroup` flag is always the opposite of `terminateIfBusy`. If they're addressing the same underlying problem, could we consolidate them and just check for a running task (and delete the task group check)?

Reply:
I looked into this a bit and I agree with you that we shouldn't need to check both of them